### PR TITLE
[history] update options for createMemoryHistory method

### PR DESCRIPTION
**Pull request by martin-badin**
Documentation/Source files:
https://github.com/remix-run/history/blob/v4/modules/createMemoryHistory.js#L39
https://github.com/remix-run/history/blob/v4/modules/LocationUtils.js#L12-L33
Discussion: #58209

Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [ ] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `npm test histor`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: <https://github.com/remix-run/history/blob/v4/modules/createMemoryHistory.js>
- [ ] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header.

### CODE REVIEW

**Comment by martin-badin** [comment]
@peterblazejewicz Please, could you review? Then I remove the history package from DT.

File: types/history/history-tests.ts [hunk 1]
```diff
@@ -62,7 +62,9 @@ let input = { value: '' };
 }
 
 {
-    let history: MemoryHistory<{ the: 'state' }> = createMemoryHistory();
+    let history: MemoryHistory<{ the: 'state' }> = createMemoryHistory({
+        initialEntries: ['/', { pathname: 'pathname' }],
```
Comment (peterblazejewicz) [inline types/history/history-tests.ts:66]: this is wrong, that object is expected to have 'key' according to a snip you've posted, `entry.key`.

Comment (martin-badin) [reply types/history/history-tests.ts:66]: not just key. The object is the same as the location. Please see this URL. The `path` object is the same as the location.

https://github.com/remix-run/history/blob/1b3ae1b5abda4fbb754c869737d02baaad358194/modules/LocationUtils.js#L14-L33

**Review by peterblazejewicz — APPROVED** [review]

**Comment by martin-badin** [comment]
Ready to merge

**Pull request merged**